### PR TITLE
feat(common): unit tests for authentication helpers

### PR DESCRIPTION
**Pull request by anwarulislam**
This pull request introduces comprehensive unit tests for various authentication mechanisms in the `hoppscotch-common` package. These tests ensure that the authentication headers and parameters are generated correctly for different scenarios, including handling template strings, optional parameters, and edge cases like empty values.

### CODE REVIEW

**Review by jamesgeorge007 — CHANGES_REQUESTED** [review]
The helper functions like `generateApiKeyAuthHeaders()` seem to take in a `request` argument, but don't consume it. Could you evaluate and remove the same if not necessary?
Also helps reduce the surface area of the test suite in context.

**Review by jamesgeorge007 — APPROVED** [review]

File: packages/hoppscotch-common/src/helpers/auth/types/__tests__/aws-signature.spec.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import { describe, expect, test, vi, beforeEach, afterEach } from "vitest"
-import { makeRESTRequest } from "@hoppscotch/data"
+import type { Environment, HoppRESTAuth } from "@hoppscotch/data"
+import { afterEach, beforeEach, describe, expect, test, vi } from "vitest"
```
[the remark this answers is not in the document]
Comment (jamesgeorge007) [reply packages/hoppscotch-common/src/helpers/auth/types/__tests__/aws-signature.spec.ts:2]: There is no local helper function anymore, and it is consumed from `test-utils`.

[the remark this answers is not in the document]
Comment (jamesgeorge007) [reply packages/hoppscotch-common/src/helpers/auth/types/__tests__/aws-signature.spec.ts:2]: There is no local helper function in context anymore, and it is consumed from test-utils.
